src/storage/src/controller.rs

```diff
@@ -1456,7 +1457,10 @@ mod persist_read_handles {
                             let futs = FuturesUnordered::new();
                             for (_id, read) in read_handles.iter_mut() {
                                 if let Some(read) = read.as_mut() {
-                                    futs.push(read.maybe_heartbeat_reader());
+                                    // downgrade_since is idempotent, so downgrading to the minimum
+                                    // element is a no-op and can be used as a lease liveness
+                                    // heartbeat.
+                                    futs.push(read.maybe_downgrade_since(&antichain_minimum_element));
```

Comment:
I wonder if we want this to be `maybe_downgrade_since_and_seqno`, I find the desired behavior here very subtle -- intentionally calling `downgrade_since` to not advance `since` but progress `seqno`

Reply:
I feel that seqno as a concept shouldn't be exposed to persist users. Maybe the real answer is reintroducing a public `maybe_heartbeat_reader` that internally has these semantics? I'd like to keep this diff small and obvious though, given how close we are to cutting this week's release. How about I leave that as a TODO?

Reply:
TODO is fine, just something to capture what's actually happening here

Reply:
done!